File: app/src/main/java/jp/juggler/subwaytooter/api/entity/TootError.java
package jp.juggler.subwaytooter.api.entity;

import android.support.annotation.Nullable;

import org.json.JSONObject;

import jp.juggler.subwaytooter.util.LogCategory;
import jp.juggler.subwaytooter.util.Utils;

public class TootError {
	private static final LogCategory log = new LogCategory( "TootError" );
	
	//	A textual description of the error
	public String error;
	
	@Nullable
	public static TootError parse( JSONObject src ){
		if( src == null ) return null;
		try{
			TootError dst = new TootError();
			dst.error = Utils.optStringX( src, "error" );
			return dst;
		}catch( Throwable ex ){
			log.trace( ex );
			log.e( ex, "TootError.parse failed." );
			return null;
		}
	}
	
}

File: app/src/main/java/jp/juggler/subwaytooter/api/entity/TootError.kt
package jp.juggler.subwaytooter.api.entity

import org.json.JSONObject

import jp.juggler.subwaytooter.util.LogCategory
import jp.juggler.subwaytooter.util.Utils

class TootError(src:JSONObject) {
	
	//	A textual description of the error
	val error : String
	
	init{
		this.error = Utils.optStringX(src, "error") ?: ""
	}

	companion object {
		private val log = LogCategory("TootError")
		
		fun parse(src : JSONObject?) : TootError? {
			if(src == null) return null
			return try {
				TootError(src)
			} catch(ex : Throwable) {
				log.trace(ex)
				log.e(ex, "TootError.parse failed.")
				null
			}
			
		}
	}
	
}
